src/plugins/intel_gpu/src/runtime/ze/ze_device.cpp

```diff
@@ -0,0 +1,373 @@
+    bool supports_pci_properties = supports_extension(extensions, ZE_PCI_PROPERTIES_EXT_NAME, ZE_PCI_PROPERTIES_EXT_VERSION_1_0);
+    bool supports_cb_events = supports_extension(extensions, ZEX_COUNTER_BASED_EVENT_EXT_NAME, ZEX_COUNTER_BASED_EVENT_VERSION_1_0);
+    bool supports_cp_offload =
+        supports_extension(extensions, ZEX_INTEL_QUEUE_COPY_OPERATIONS_OFFLOAD_HINT_EXP_NAME, ZEX_INTEL_QUEUE_COPY_OPERATIONS_OFFLOAD_HINT_EXP_VERSION_1_0);
+    bool supports_dp_properties =
+        supports_extension(extensions, ZE_INTEL_DEVICE_MODULE_DP_PROPERTIES_EXP_NAME, ZE_INTEL_DEVICE_MODULE_DP_PROPERTIES_EXP_VERSION_1_0);
+
+    void *device_properties_next = nullptr;
+    ze_device_ip_version_ext_t ip_version_properties = {ZE_STRUCTURE_TYPE_DEVICE_IP_VERSION_EXT, device_properties_next, 0};
+    if (supports_ip_version) {
+        device_properties_next = &ip_version_properties;
+    }
+    ze_device_properties_t device_properties{ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES_1_2, device_properties_next};
+    OV_ZE_EXPECT(zeDeviceGetProperties(device, &device_properties));
+
+    ze_device_compute_properties_t device_compute_properties{ZE_STRUCTURE_TYPE_DEVICE_COMPUTE_PROPERTIES};
+    OV_ZE_EXPECT(zeDeviceGetComputeProperties(device, &device_compute_properties));
+
+    uint32_t queue_properties_count = 0;
+    OV_ZE_EXPECT(zeDeviceGetCommandQueueGroupProperties(device, &queue_properties_count, nullptr));
+
+    std::vector<ze_command_queue_group_properties_t> queue_properties(queue_properties_count);
+    for (auto& mp : queue_properties) {
+        mp.stype = ZE_STRUCTURE_TYPE_COMMAND_QUEUE_GROUP_PROPERTIES;
+    }
+
+    OV_ZE_EXPECT(zeDeviceGetCommandQueueGroupProperties(device, &queue_properties_count, &queue_properties[0]));
+
+    auto compute_queue_props = std::find_if(queue_properties.begin(), queue_properties.end(), [](const ze_command_queue_group_properties_t& qp) {
+        return (qp.flags & ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) != 0;
+    });
+
+    OPENVINO_ASSERT(compute_queue_props != queue_properties.end());
+
+    uint32_t memory_properties_count = 0;
+    OV_ZE_EXPECT(zeDeviceGetMemoryProperties(device, &memory_properties_count, nullptr));
+
+    std::vector<ze_device_memory_properties_t> device_memory_properties(memory_properties_count);
+    for (auto& mp : device_memory_properties) {
+        mp.stype = ZE_STRUCTURE_TYPE_DEVICE_MEMORY_PROPERTIES;
+    }
+    OV_ZE_EXPECT(zeDeviceGetMemoryProperties(device, &memory_properties_count, &device_memory_properties[0]));
+
+    ze_device_memory_access_properties_t device_memory_access_properties{ZE_STRUCTURE_TYPE_DEVICE_MEMORY_ACCESS_PROPERTIES};
+    OV_ZE_EXPECT(zeDeviceGetMemoryAccessProperties(device, &device_memory_access_properties));
+
+    auto mem_properties = std::find_if(device_memory_properties.begin(), device_memory_properties.end(), [](const ze_device_memory_properties_t& p) {
+        auto name = std::string(p.name);
+        return name == "DDR" || name == "HBM";
```

Comment:
* out of curiosity, what other name does it have other than DDR and HBM? e.g. does it have property like "SRAM"?
* Am I correct that intel GPUs have only 1 type of memory?